pkg/node/node.go

```diff
@@ -250,9 +254,102 @@ func (n *Node) publishSignerPublicKey(
 		return fmt.Errorf("failed to submit public key: [%v]", err)
 	}
 
+	// Trigger the member public key submission monitoring as separate goroutine.
+	// This way the keygen result is decoupled from the eventual outcome
+	// of the `SubmitKeepPublicKey` transaction. This is safer because
+	// keygen process will return an error only in case when the submission
+	// reverts at the dry-run stage. In other cases, keygen will return
+	// and persist the signer (and its key shares) and will treat the keep
+	// as normally operating which gives more chances to avoid serious errors.
+	go n.monitorMemberPublicKeySubmission(keepAddress, publicKey)
+
 	return nil
 }
 
+func (n *Node) monitorMemberPublicKeySubmission(
+	keepAddress common.Address,
+	publicKey [64]byte,
+) {
+	const maxIterations = 3
+
+	for iteration := 1; iteration <= maxIterations; iteration++ {
+		logger.Infof(
+			"starting iteration [%v] of member public key "+
+				"submission monitoring for keep: [%v]",
+			iteration,
+			keepAddress.Hex(),
+		)
+
+		currentBlock, err := n.ethereumChain.BlockCounter().CurrentBlock()
```

Comment:
Why don't we use `WaitForBlockConfirmations` from `keep-common` for all of this? The action is public key resubmission.